### PR TITLE
FE: Dashboard Skeleton

### LINKED ISSUE #17: FE: Dashboard Skeleton
- Adapt the current project skeleton to match project mockups
- The project skeleton will have the general frontend structure of the project, but without the specific components filled out
- If there is a navbar in the mockups, the skeleton should include the navbar + content pages for the app for a logged in user experience
- The navbar should be persistent across the whole logged in user experience and include a placeholder image for the user
- Dropdown for profile in navbar should show two links:
    - logout (already implemented on the backend)
    - go to profile

### PR DESCRIPTION
**Pull request by navjot-chahal**
### What this PR does:
- Build Wireframe for the Navbar for the logged-in session and otherwise.
- Add routes and dummy pages for the `Profile/`, `/messages`, `My Jobs/` and `My Sitters` section.

### Screenshots:
- Landing page view.

Landing page

- Logged-in page view.

Logged in

Sitters

-Mobile view (Needs future improvements!)

Mobile view

### Any information needed to test this feature:
- Setup proper database (local or cloud).
- Run the Express server from the `server` folder with the `npm run dev` command.
- Make sure proper environment variables are set up.
- Signup as a new user.

closes #17

### CODE REVIEW

File: client/src/pages/Messages/Messages.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const Profile = (): JSX.Element => {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+  }
```
Comment (ezeilo-su) [inline client/src/pages/Messages/Messages.tsx:16]: :+1:

File: client/src/pages/MyJobs/MyJobs.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const Profile = (): JSX.Element => {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+  }
```
Comment (ezeilo-su) [inline client/src/pages/MyJobs/MyJobs.tsx:16]: :+1:

File: client/src/pages/MySitters/MySitters.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+const Profile = (): JSX.Element => {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+  }
```
Comment (ezeilo-su) [inline client/src/pages/MySitters/MySitters.tsx:16]: :+1:

File: client/src/pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const Profile = (): JSX.Element => {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+  }
```
Comment (ezeilo-su) [inline client/src/pages/Profile/Profile.tsx:16]: :+1:

File: client/src/components/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+  useEffect(() => {
+    initSocket();
+  }, [initSocket]);
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+
+  return (
+    <>
+      <CssBaseline />
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:112]: Please remove the <CssBaseline /> here as it is a global reset component and only needs to be included once, ideally in the 'App.tsx'. If you still intend to use a baseline for the children, look into the usage of a 'ScopedCssBaseline'

Ref: https://mui.com/api/scoped-css-baseline/

File: client/src/pages/MyJobs/MyJobs.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import useStyles from './useStyles';
+
+const Profile = (): JSX.Element => {
+  const classes = useStyles();
+  const { loggedInUser } = useAuth();
+  const history = useHistory();
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
```
Comment (rajivtitus) [inline client/src/pages/MyJobs/MyJobs.tsx:14]: Same as comment above!

File: client/src/components/NavBar/NavBar.tsx [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            </Typography>
+            <Notifications className={`${classes.desktopView}`} />
+          </Badge>
+        </Button>
+      </NavLink>
+      {loggedInUser.isSitter ? (
+        <NavLink to="/my-jobs" className={`${classes.link} ${classes.mobileView}`}>
+          <Button className={classes.menuBarButton}>
+            <Typography variant="h6" color="textPrimary" className={classes.menuBarText}>
+              My Jobs
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:47]:
```suggestion
              {loggedInUser.isSitter ? "My Jobs" : "My Sitters"
```

File: client/src/components/NavBar/NavBar.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+          <Badge color="primary" variant="dot">
+            <Typography variant="h6" color="textPrimary" className={`${classes.menuBarText} ${classes.mobileView}`}>
+              Notifications
+            </Typography>
+            <Notifications className={`${classes.desktopView}`} />
+          </Badge>
+        </Button>
+      </NavLink>
+      {loggedInUser.isSitter ? (
+        <NavLink to="/my-jobs" className={`${classes.link} ${classes.mobileView}`}>
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:44]:
```suggestion
        <NavLink to={loggedInUser.isSitter ? : "/my-jobs" : "/my-sitters"} className={`${classes.link} ${classes.mobileView}`}>
```

File: client/src/components/NavBar/NavBar.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+import AuthMenu from '../AuthMenu/AuthMenu';
+import Logo from '../../Images/logo.png';
+
+// Styles
+import useStyles from './useStyles';
+
+function LoggedInBar({ classes, loggedInUser }: any): JSX.Element {
+  return (
+    <Box>
+      <NavLink to="/dashboard" className={classes.link}>
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:33]: You can combine react-router's Link with the MUI Button :)

Ref: https://stackoverflow.com/questions/51642532/how-to-make-a-material-ui-react-button-act-as-a-react-router-dom-link

**Comment by navjot-chahal** [comment]
@rajivtitus Please review these Hotfix changes in commit `Changes-1`. Thanks for a detailed review of my PR! 😀

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes 💯

**Pull request merged**